libbeat/outputs/elasticsearch/client.go

```diff
@@ -189,8 +189,25 @@ func (client *Client) Publish(ctx context.Context, batch publisher.Batch) error
 	rest, err := client.publishEvents(ctx, events)
 
 	switch {
-	case err == errPayloadTooLarge:
-		batch.Drop()
+	case errors.Is(err, errPayloadTooLarge):
+		if batch.SplitRetry() {
+			// If we successfully split the batch, we report the events to the
+			// observer as "failed", since this indicates a publish error that
+			// will be retried.
+			client.observer.Failed(len(events))
```

Comment:
Is it worth adding a metric like `splits` for the number of split batches? Do we have any way to know if batch splitting is happening or possibly causing problems?

Probably to debug any issues from batch splitting we'd need to look at the events and batches directly, but having a counter or some way to know if it has ever happened would give us a quick way to rule it out.

Reply:
Done, added new metric `batches.split` alongside `events.{failed, dropped, etc}`.